Json.More/JsonSerializerOptionsExtensions.cs

```diff
@@ -60,11 +60,11 @@ public static JsonConverter<T> GetConverter<T>(this JsonSerializerOptions option
 	/// <param name="value">The value to serialize.</param>
 	/// <param name="typeInfo">An explicit typeInfo to use for looking up the Converter. If not provided, options.GetTypeInfo will be used.</param>
 	/// <returns>The value that was converted.</returns>
-	public static void Write<T>(this JsonSerializerOptions options, Utf8JsonWriter writer, T? value, JsonTypeInfo<T>? typeInfo = null)
+	[UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "Serialize is safe in AOT if the JsonSerializerOptions come from the source generator. Requiring the JsonTypeInfo parameter helps enforce that.")]
+	[UnconditionalSuppressMessage("AOT", "IL3050:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "Serialize is safe in AOT if the JsonSerializerOptions come from the source generator. Requiring the JsonTypeInfo parameter helps enforce that.")]
+	public static void Write<T>(this JsonSerializerOptions options, Utf8JsonWriter writer, T? value, JsonTypeInfo<T>? typeInfo)
 	{
-#pragma warning disable IL2026, IL3050 // This helper is expected to be called with an options object that covers the needed TypeInfos.
```

Comment:
If we're not using the type info, why include it as a parameter?

Reply:
For static validation. There seems to be no way to bind a TypeInfo to a different Options than the one JsonSerializerContext that it was born with. Using the TypeInfo ignores any other types in the provided options context, so we prefer the options context.

@eiriktsarpalis this goes to the JsonSerializerContext customization I was asking you for in chat.

Reply:
All that's fair, but if we're not using a parameter, why not remove it?

Reply:
But the `.Read<T>()` method actually uses the parameter to get the appropriate converter.  The `.Write<T>()` method doesn't use the parameter.

Secondly, `.Write<T>()` includes a `T value` parameter that informs the generic type.

Reply:
If the `.Write<T>()` method is supposed to mirror the `.Read<T>()` method, I recommend changing the body to

```c#
options.GetConverter<T>(typeInfo).Write(writer, value!, options);
```

Reply:
Sure, I'd been wanting to do this anyway so let me make that change and see how the tests feel about it.

Reply:
(probably remove the `= null` from the parameter, too.)

Reply:
It works! Updated.

Reply:
Do we still need the attributes now?

Reply:
Good call. Removed.

Reply:
Actually, wasn't GetConverter<T>() there as a public extension method before? I had the = null as compatibility for external callers. LMK if you think I leave it or bring it back.

Reply:
It's not optional on `.Read<T>()`.  I say we keep it off.

Reply:
Passing both an `options` and `typeInfo` parameters are redundant and could in fact be mutually contradictory (e.g. because `typeInfo` encapsulates a different options instance). If the motivation is to avoid the linker warning it seems to me that calling
```C#
var typeInfo = (JsonTypeInfo<T>?)options.GetTypeInfo(typeof(T));
if (typeInfo is null) throw new Exception(); /* some error that T is not defined in the options */
return JsonSerializer.Serializer(writer, value, typeInfo);
```

Reply:
The goal is that we want to include the type info while also respecting whatever options the user sends us (e.g. write indented, name casing, etc).